examples/vision/reptile.py

```diff
@@ -0,0 +1,248 @@
+            self.data[label].append(image)
+            self.labels = list(self.data.keys())
+
+    def get_mini_dataset(
+        self, batch_size, repetitions, shots, num_classes, split=False
+    ):
+
+        temp_labels = np.zeros(shape=(num_classes * shots))
+        temp_images = np.zeros(shape=(num_classes * shots, 28, 28, 1))
+        if split:
+            test_labels = np.zeros(shape=(num_classes))
+            test_images = np.zeros(shape=(num_classes, 28, 28, 1))
+        # Get a random subset of labels from the entire label set.
+        label_subset = random.choices(self.labels, k=num_classes)
+        for class_idx, class_obj in enumerate(label_subset):
+            # Use enumerated index value as a temporary label for mini_batch in
+            # few shot learning.
+            temp_labels[class_idx * shots : (class_idx + 1) * shots] = class_idx
+            # If creating a split dataset for testing, select an extra sample from each
+            # label to create the test dataset.
+            if split:
+                test_labels[class_idx] = class_idx
+                images_to_split = random.choices(
+                    self.data[label_subset[class_idx]], k=shots + 1
+                )
+                test_images[class_idx] = images_to_split[-1]
+                temp_images[
+                    class_idx * shots : (class_idx + 1) * shots
+                ] = images_to_split[:-1]
+            else:
+                # For each index in the randomly selected label_subset, sample the
+                # necessary number of images.
+                temp_images[
+                    class_idx * shots : (class_idx + 1) * shots
+                ] = random.choices(self.data[label_subset[class_idx]], k=shots)
+
+        dataset = tf.data.Dataset.from_tensor_slices(
+            (temp_images.astype(np.float32), temp_labels.astype(np.int32))
+        )
+        dataset = dataset.shuffle(100).batch(batch_size).repeat(repetitions)
+        if split:
+            return dataset, test_images, test_labels
+        return dataset
+
+
+import urllib3
+
+urllib3.disable_warnings()
+train_dataset = Dataset(training=True)
+test_dataset = Dataset(training=False)
+
+"""
+## Build the Model
+"""
+
+
+class ModelLayer(layers.Layer):
+    def __init__(self, **kwargs):
+        super(ModelLayer, self).__init__(**kwargs)
+        self.conv = layers.Conv2D(filters=64, kernel_size=3, strides=2, padding="same")
+        self.bnorm = layers.BatchNormalization()
+        self.activation = layers.ReLU()
+
+    def call(self, inputs):
+        x = self.conv(inputs)
+        x = self.bnorm(x)
+        return self.activation(x)
+
+
+inputs = layers.Input(shape=(28, 28, 1))
+layer1 = ModelLayer()(inputs)
+layer2 = ModelLayer()(layer1)
+layer3 = ModelLayer()(layer2)
+layer4 = ModelLayer()(layer3)
+flatten = layers.Flatten()(layer4)
+dense = layers.Dense(classes, activation="softmax")(flatten)
+# Log softmax
+outputs = tf.math.log(layers.Softmax()(dense))
```

Comment:
Taking the log of a softmax will lead to numerical instability. Either use softmax and train with `CategoricalCrossentropy()`, or do not apply any activation and train with `CategoricalCrossentropy(from_logits=True)`